parts/k8s/kuberneteswindowsfunctions.ps1

```diff
@@ -96,6 +96,36 @@ function Retry-Command
     }
 }
 
+function Invoke-Executable
+{
+    Param(
+        [string]
+        $Executable,
+        [string[]]
+        $ArgList,
+        [int[]]
+        $AllowedExitCodes = @(0),
+        [int]
+        $Retries = 1,
+        [int]
+        $RetryDelaySeconds = 1
+    )
+
+    for ($i = 0; $i -lt $Retries; $i++) {
+        Write-Log "Running $Executable $ArgList ..."
+        & $Executable $ArgList
+        if ($LASTEXITCODE -notin $AllowedExitCodes) {
+            Write-Log "$Executable returned unsuccessfully with exit code $LASTEXITCODE"
+            continue
+        } else {
+            Write-Log "$Executable returned successfully"
+            return
+        }
+    }
+
+    throw "Exhasted retries for $Executable $ArgList"

```

Comment:
s/Exhasted/Exhausted/

Reply:
Fixed (but had to rebase so comments are stale)